src/backend/distributed/planner/multi_explain.c

```diff
@@ -124,85 +118,54 @@ MultiExplainOneQuery(Query *query, IntoClause *into, ExplainState *es,
+	INSTR_TIME_SUBTRACT(planDuration, planStart);
 
+	/* if not a distributed query, use plain explain infrastructure */
+	if (!HasCitusToplevelNode(plan))
+	{
 		/* run it (if needed) and produce output */
 		ExplainOnePlan(plan, into, es, queryString, params, &planDuration);
-
-		return;
-	}
-
-	/*
-	 * standard_planner scribbles on it's input, but for deparsing we need the
-	 * unmodified form. So copy once we're sure it's a distributed query.
-	 */
-	originalQuery = copyObject(query);
-
-	/* measure the full planning time to display in EXPLAIN ANALYZE */
-	INSTR_TIME_SET_CURRENT(planStart);
-
-	restrictionContext = CreateAndPushRestrictionContext();
-
-	PG_TRY();
-	{
-		/* call standard planner to modify the query structure before multi planning */
-		initialPlan = standard_planner(query, cursorOptions, params);
-
-		commandType = initialPlan->commandType;
-		if (commandType == CMD_INSERT || commandType == CMD_UPDATE ||
-			commandType == CMD_DELETE)
-		{
-			if (es->analyze)
-			{
-				ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-								errmsg("Using ANALYZE for INSERT/UPDATE/DELETE on "
-									   "distributed tables is not supported.")));
-			}
-		}
-
-		multiPlan = CreatePhysicalPlan(originalQuery, query, restrictionContext);
 	}
-	PG_CATCH();
+	else
 	{
-		PopRestrictionContext();
-		PG_RE_THROW();
+		MultiExplainOnePlan(plan, into, es, queryString, params, &planDuration);
 	}
-	PG_END_TRY();
+}
 
-	PopRestrictionContext();
 
-	INSTR_TIME_SET_CURRENT(planDuration);
-	INSTR_TIME_SUBTRACT(planDuration, planStart);
+/*
+ * MultiExplainOnePlan explains the plan for an individual distributed query.
+ */
+static void
+MultiExplainOnePlan(PlannedStmt *plan, IntoClause *into,
+					ExplainState *es, const char *queryString,
+					ParamListInfo params, const instr_time *planDuration)
+{
+	MultiPlan *multiPlan = NULL;
+	CmdType commandType = CMD_UNKNOWN;
+	Job *workerJob = NULL;
+	bool routerExecutablePlan = false;
 
-	if (ExplainMultiLogicalPlan)
+	commandType = plan->commandType;
+	if (commandType == CMD_INSERT || commandType == CMD_UPDATE ||
+		commandType == CMD_DELETE)
 	{
-		MultiTreeRoot *multiTree = MultiLogicalPlanCreate(query);
-		char *logicalPlanString = CitusNodeToString(multiTree);
-		char *formattedPlanString = pretty_format_node_dump(logicalPlanString);
-
-		appendStringInfo(es->str, "logical plan:\n");
-		appendStringInfo(es->str, "%s\n", formattedPlanString);
+		if (es->analyze)
+		{
+			ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+							errmsg("Using ANALYZE for INSERT/UPDATE/DELETE on "
+								   "distributed tables is not supported.")));
+		}
 	}
 
-	if (ExplainMultiPhysicalPlan)
-	{
-		char *physicalPlanString = CitusNodeToString(multiPlan);
-		char *formattedPlanString = pretty_format_node_dump(physicalPlanString);
-
-		appendStringInfo(es->str, "physical plan:\n");
-		appendStringInfo(es->str, "%s\n", formattedPlanString);
-	}
+	multiPlan = GetMultiPlan(plan);
 
 	if (!ExplainDistributedQueries)
```

Comment:
I never really understood why this was behind a GUC. Know why we can't just enable this always?